### PR TITLE
Part of #20163 for file ui\components\app\terms-of-use-popup\terms-of-use-popup.js

### PR DESCRIPTION
**Pull request by PrgrmrHarshShukla**
## Explanation
This PR is a part of the issue #20163 Replace deprecated CheckBox component with Checkbox from the component-library for file
`ui\components\app\terms-of-use-popup\terms-of-use-popup.js`

Also fixed a small typo in 
`ui\components\app\snaps\install-error\install-error.stories.js`

### Before
Screenshot 2023-08-25 075322

### After

Screenshot 2023-08-25 075410

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by PrgrmrHarshShukla** [comment]
Hey @garrettbear and @georgewrmarshall ,

Kindly look at the positioning of the checkbox and label text.
There is some change in their positioning in the before and after UI.
I thought it would be better to ask if it should be made to follow the earlier positioning.

Also,
the label text was bold earlier.
I could not make out how to do so, please help me with some idea on how to do that, in case it is needed.

**Comment by PrgrmrHarshShukla** [comment]
Also,
please let me know if I should move the typo problem to a new PR to keep things separate.
I thought it is a relatively small problem, so it might be okay to keep it here.

Regards,
Harsh

**Review by georgewrmarshall — COMMENTED** [review]
Hey @PrgrmrHarshShukla, these changes look good! Great call out on the visual regressions! This is an issue with the Checkbox component that can be addressed separately. There are some failing e2e tests that will need to be resolved for the PR to get merged. If you can kindly see to those that would be great!

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
Both the e2e tests are failing because of the same reason.
The reason being `TimeoutError` while waiting for an element ( probably  `Checkbox` element, we can get a hint here `[data-testid=&#x22;terms-of-use-checkbox&#x22;]` below ) to be located by the CSS selector.

Screenshot 2023-08-29 072550

Screenshot 2023-08-29 072622



## Also
while updating the snapshots for another PR #20535 , I encountered this,


Screenshot 2023-08-29 072650


Probably, these are connected in some way.


## Possible Solution

Probably we can let the CSS selector grab the just higher level component of the `Checkbox`, 
that is, the **div** with `className="create-new-vault__terms"`.
image

Also,
here the `Checkbox` is rendered conditionally and the condition `includeTerms` is initially false, this might be making it unavailable for the CSS selector to grab.


🙏🏼
@georgewrmarshall , please share your insights on this.

Regards,
Harsh

File: ui/components/app/terms-of-use-popup/terms-of-use-popup.js [hunk 1]
```diff
@@ -1162,26 +1161,23 @@ export default function TermsOfUsePopup({ onAccept }) {
...
             alignItems={AlignItems.flexStart}
             marginLeft={3}
             marginRight={3}
             gap={2}
           >
-            <CheckBox
+            <Checkbox
               id="terms-of-use__checkbox"
               className="terms-of-use__checkbox"
               dataTestId="terms-of-use-checkbox"
```
Comment (georgewrmarshall) [inline ui/components/app/terms-of-use-popup/terms-of-use-popup.js:1173]: Update to
```suggestion
              data-testid="terms-of-use-checkbox"
```

File: ui/components/app/terms-of-use-popup/terms-of-use-popup.js [hunk 2]
```diff
@@ -1162,26 +1161,23 @@ export default function TermsOfUsePopup({ onAccept }) {
...
-              checked={isTermsOfUseChecked}
+              isChecked={isTermsOfUseChecked}
               onClick={() => {
```
Comment (georgewrmarshall) [inline ui/components/app/terms-of-use-popup/terms-of-use-popup.js:1175]: Update to
```suggestion
              onChange={() => {
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Left some suggestions

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Not sure why the e2e tests are failing here will try updating from develop

**Comment by PrgrmrHarshShukla** [comment]
Hey @garrettbear ,
Your help is probably needed here.

**Review by garrettbear — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]
LGTM ✅ 
- checked storybook

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**